libtock/tock.c

```diff
@@ -277,19 +322,43 @@ void yield(void) {
     task_cur = (task_cur + 1) % TASK_QUEUE_SIZE;
     task.cb(task.arg0, task.arg1, task.arg2, task.ud);
   } else {
+    register uint32_t a0  asm ("a0") = 1; // yield-wait
     asm volatile (
-      "li    a4, 0\n"
+      "li    a5, 0\n"
       "ecall\n"
       :
-      :
-      : "memory", "a0", "a1", "a2", "a3", "a4", "a5", "a6", "a7",
+      : "r" (a0)
+      : "memory", "a1", "a2", "a3", "a4", "a6", "a7",
```

Comment:
doesn't this still need to mark a5 as clobbered? Especially now that this snippet directly overwrites it even before the `ecall`?

Reply:
Yes! Nice catch. I have no idea why that happened.